src/ja/common/job.py

```diff
@@ -146,22 +204,55 @@ def status(self, new_status: JobStatus) -> None:
 
         @param new_status The new status of the job.
         """
+        self._status = new_status
 
     @property
-    def docker_context(self) -> IDockerContext:
-        """!
-        @return The docker context of the job.
-        """
-
-    @property
-    def docker_constraints(self) -> DockerConstraints:
+    def label(self) -> str:
         """!
-        @return The docker constraints of the job.
+        @return: The label set by the user for this job.
         """
+        return self._label
 
     def to_dict(self) -> Dict[str, object]:
-        pass
+        _dict: Dict[str, object] = dict()
+        _dict["uid"] = self.uid
+        _dict["owner_id"] = self.owner_id
+        _dict["email"] = self.email
+        _dict["scheduling_constraints"] = self.scheduling_constraints.to_dict()
+        _dict["docker_context"] = self.docker_context.to_dict()
+        _dict["docker_constraints"] = self.docker_constraints.to_dict()
+        _dict["status"] = self.status
+        _dict["label"] = self.label
+        return _dict
 
     @classmethod
     def from_dict(cls, property_dict: Dict[str, object]) -> "Job":
-        pass
+        _uid = cls._get_str_from_dict(property_dict=property_dict, key="uid", mandatory=False)
```

Comment:
Shouldn't we specify the type of these local variables?
`_uid: str = cls._get_str_from_dict(property_dict=property_dict, key="uid", mandatory=False)`

Reply:
Because `_get_str_from_dict` always returns an object of type str the type of `_uid` is already known to be str.
For this reason type hints are optional here.
Do you think the code would be easier to understand if there were type hints?

Reply:
No, it wouldn't improve readability, but I thought that typing is always mandatory in PSE.

> Because _get_str_from_dict always returns an object of type str the type of _uid is already known to be str.
In most strongly typed languages, you'd have to specify the type nevertheless.

Let's just wait for more opinions...

Reply:
I think it is fine, no one will care much about this, outside of us, so readability is more important.